### DISCUSSION TITLE
Getting classes

### DISCUSSION THREAD

**Discussion by blacksmith001** [ultralytics/ultralytics#17218] [Docs]
I want to process the results of my classification. I was trying to get classes like this:

for result in results:
    boxes = result.boxes  # Get the boxes for the first result
    classes = result.boxes.cls  # Get class indices

But then I am getting:
" 'NoneType' object has no attribute 'cls' "

**Comment by UltralyticsAssistant** [comment]
👋 Hello @blacksmith001, thank you for reaching out to the Ultralytics community 🚀! It sounds like you're encountering an issue with processing classification results.

If this is a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) that we can use to help debug the problem? This will greatly assist us in identifying and resolving the issue.

For issues related specifically to deriving class indices, it might be helpful to ensure your implementation aligns with our [Python usage examples](https://docs.ultralytics.com/usage/python/). The error `'NoneType' object has no attribute 'cls'` suggests that the result may not contain any detected boxes, so it could be useful to verify the outputs and check for empty results.

If you need further assistance, an Ultralytics engineer will look into this soon. In the meantime, you might find our community a valuable resource. For real-time discussions, consider joining the conversation on [Discord](https://ultralytics.com/discord) 🎧, or explore [Discourse](https://community.ultralytics.com) and our [Subreddit](https://reddit.com/r/ultralytics) for more in-depth discussions.

## Upgrade

To ensure you have all the latest features and fixes, upgrade to the newest version of the `ultralytics` package:

```bash
pip install -U ultralytics
```

## Environments

For guaranteed compatibility and ease of use, you can run YOLO in one of our verified environments, each preinstalled with necessary dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

With all Continuous Integration (CI) tests passing, the green badge indicates that our YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are verified for correct operation across various platforms.

**Comment by glenn-jocher** [comment]
@blacksmith001 it seems like the `boxes` attribute might be `None`. Ensure your model is set up for detection tasks, as classification results are accessed via `result.probs` for class probabilities.
